tests/components/vizio/test_media_player.py

```diff
@@ -85,154 +79,161 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-async def _test_setup(
-    hass: HomeAssistantType, ha_device_class: str, vizio_power_state: Optional[bool]
+async def _add_config_entry_to_hass(
+    hass: HomeAssistantType, config_entry: MockConfigEntry
 ) -> None:
-    """Test Vizio Device entity setup."""
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+
+def _get_ha_power_state(vizio_power_state: Optional[bool]) -> str:
+    """Return HA power state given Vizio power state."""
     if vizio_power_state:
-        ha_power_state = STATE_ON
-    elif vizio_power_state is False:
-        ha_power_state = STATE_OFF
-    else:
-        ha_power_state = STATE_UNAVAILABLE
-
-    if ha_device_class == DEVICE_CLASS_SPEAKER:
-        vizio_device_class = VIZIO_DEVICE_CLASS_SPEAKER
-        config_entry = MockConfigEntry(
-            domain=DOMAIN,
-            data=vol.Schema(VIZIO_SCHEMA)(MOCK_SPEAKER_CONFIG),
-            unique_id=UNIQUE_ID,
-        )
-        dict_to_return = {
-            "volume": int(MAX_VOLUME[vizio_device_class] / 2),
-            "mute": "Off",
-            "eq": CURRENT_EQ,
-        }
-    else:
-        vizio_device_class = VIZIO_DEVICE_CLASS_TV
-        config_entry = MockConfigEntry(
-            domain=DOMAIN,
-            data=vol.Schema(VIZIO_SCHEMA)(MOCK_USER_VALID_TV_CONFIG),
-            unique_id=UNIQUE_ID,
-        )
-        dict_to_return = {
-            "volume": int(MAX_VOLUME[vizio_device_class] / 2),
-            "mute": "Off",
-        }
+        return STATE_ON
+
+    if vizio_power_state is False:
+        return STATE_OFF
+
+    return STATE_UNAVAILABLE
+
+
+def _assert_sources_and_volume(attr: Dict[str, Any], vizio_device_class: str) -> None:
+    """Assert source list, source, and volume level based on attr dict and device class."""
+    assert attr["source_list"] == INPUT_LIST
+    assert attr["source"] == CURRENT_INPUT
+    assert (
+        attr["volume_level"]
+        == float(int(MAX_VOLUME[vizio_device_class] / 2))
+        / MAX_VOLUME[vizio_device_class]
+    )
+
+
+def _get_attr_and_assert_base_attr(
+    hass: HomeAssistantType, device_class: str, power_state: str
+) -> Dict[str, Any]:
+    """Return entity attributes  after asserting name, device class, and power state."""
+    attr = hass.states.get(ENTITY_ID).attributes
+    assert attr["friendly_name"] == NAME
+    assert attr["device_class"] == device_class
 
+    assert hass.states.get(ENTITY_ID).state == power_state
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5373293)